xtable-core/src/main/java/org/apache/xtable/parquet/ParquetStatsExtractor.java

```diff
@@ -132,19 +181,23 @@ public static InternalDataFile toInternalDataFile(Configuration hadoopConf, Path
     try {
       FileSystem fs = FileSystem.get(hadoopConf);
       file = fs.getFileStatus(parentPath);
-      // InputPartitionFields partitionInfo = initPartitionInfo();
       footer = parquetMetadataExtractor.readParquetMetadata(hadoopConf, parentPath);
       MessageType schema = parquetMetadataExtractor.getSchema(footer);
       columnStatsForAFile = getColumnStatsForaFile(footer);
-      // partitionValues = partitionExtractor.createPartitionValues(
-      //               partitionInfo);
+      partitionValues =
+          partitionValueExtractor.extractPartitionValues(
+              partitionSpecExtractor.spec(
+                  partitionValueExtractor.extractSchemaForParquetPartitions(
+                      parquetMetadataExtractor.readParquetMetadata(hadoopConf, file.getPath()),
+                      file.getPath().toString())),
+              parentPath.toString());
     } catch (java.io.IOException e) {
```

Comment:
I think this is just hiding the exception currently and we need to fix this.

Reply:
It catches the exception but does not throw any new exception. The user will not know of the errors.